include/GLFW/glfw3.h

```diff
@@ -1900,107 +1964,6 @@ typedef void (* GLFWmonitorfun)(GLFWmonitor* monitor, int event);
- *  axis movement callback function has the following signature:
- *  @code
- *  void function_name(int jid, int axis, float position)
- *  @endcode
- *
- *  @param[in] jid The joystick that had an axis moved.
- *  @param[in] axis The [joystick axis](@ref gamepad axes) that was moved.
- *  @param[in] position A value between -1.0 and 1.0 that indicates the position of the axis.
- *
- *  @sa @ref input_gamepad_axis
- *  @sa @ref glfwSetJoystickAxisCallback
- *
- *  @since Added in version 3.4.
- *  @ingroup input
- */
-typedef void (* GLFWjoystickaxisfun)(int,int,float);
-
-/*! @brief The function pointer type for joystick hat movement callbacks.
- *
- *  This is the function pointer type for joystick hat movement callbacks.  A joystick
- *  hat movement callback function has the following signature:
- *  @code
- *  void function_name(int jid, int hat, int position)
- *  @endcode
- *
- *  @param[in] jid The joystick that had an axis moved.
- *  @param[in] hat The [joystick hat](@ref joystick hats) that was moved.
- *  @param[in] position A value that indicates the position of the hat.
- *  The position parameter is one of the following values:
- *
- *  Name                  | Value
- *  ----                  | -----
- *  `GLFW_HAT_CENTERED`   | 0
- *  `GLFW_HAT_UP`         | 1
- *  `GLFW_HAT_RIGHT`      | 2
- *  `GLFW_HAT_DOWN`       | 4
- *  `GLFW_HAT_LEFT`       | 8
- *  `GLFW_HAT_RIGHT_UP`   | `GLFW_HAT_RIGHT` \| `GLFW_HAT_UP`
- *  `GLFW_HAT_RIGHT_DOWN` | `GLFW_HAT_RIGHT` \| `GLFW_HAT_DOWN`
- *  `GLFW_HAT_LEFT_UP`    | `GLFW_HAT_LEFT` \| `GLFW_HAT_UP`
- *  `GLFW_HAT_LEFT_DOWN`  | `GLFW_HAT_LEFT` \| `GLFW_HAT_DOWN`
- *
- *  The diagonal directions are bitwise combinations of the primary (up, right,
- *  down and left) directions and you can test for these individually by ANDing
- *  it with the corresponding direction.
- *
- *  @sa @ref input_joystick_hat
- *  @sa @ref glfwSetJoystickHatCallback
- *
- *  @since Added in version 3.4.
- *  @ingroup input
- */
-typedef void (* GLFWjoystickhatfun)(int,int,int);
-
-/*! @brief The function pointer type for game pad state changes.
- *
- *  This is the function pointer type for game pad state change callbacks.
- *  A game pad state change callback function has the following signature:
- *  @code
- *  void function_name(int jid, GLFWgamepadstate* state)
- *  @endcode
- *
- *  @param[in] jid The ID of the game pad that changed state.
- *  @param[in] buttons The states of each
- *             [gamepad button](@ref gamepad_buttons),
- *             `GLFW_PRESS` or `GLFW_RELEASE`.
- *  @param[in] axes The states of each [gamepad axis](@ref gamepad_axes),
- *                  in the range -1.0 to 1.0 inclusive.
- *
- *  @sa @ref input_gamepad
- *  @sa @ref glfwSetGamepadStateCallback
- *
- *  @since Added in version 3.4.
- *  @ingroup input
- */
-typedef void (* GLFWgamepadstatefun)(int, unsigned char buttons[15],
-                                          float axes[6]);
```

Comment:
Moved header declarations below gamepadstate def because of chagne to gamepadstatefun definition using gamepadstate as argument